src/components/AccountDrawer/PrefetchBalancesWrapper.tsx

```diff
@@ -35,36 +34,49 @@ function useHasUpdatedTx(account: string | undefined) {
   }, [account, currentChainTxs, previousPendingTxs])
 }
 
+export function useCachedPortfolioBalancesQuery({ account }: { account?: string }) {
+  return usePortfolioBalancesQuery({
+    skip: !account,
+    variables: { ownerAddress: account ?? '' },
+    fetchPolicy: 'cache-only', // PrefetchBalancesWrapper handles balance fetching/staleness; this component only reads from cache
+    errorPolicy: 'all',
+  })
+}
+
+const hasUnfetchedBalancesAtom = atom<boolean>(true)
+
 /* Prefetches & caches portfolio balances when the wrapped component is hovered or the user completes a transaction */
-export default function PrefetchBalancesWrapper({ children }: PropsWithChildren) {
+export default function PrefetchBalancesWrapper({
+  children,
+  shouldFetchOnAccountUpdate,
+}: PropsWithChildren<{ shouldFetchOnAccountUpdate: boolean }>) {
   const { account } = useWeb3React()
   const [prefetchPortfolioBalances] = usePortfolioBalancesLazyQuery()
-  const [drawerOpen] = useAccountDrawer()
 
-  const [hasUnfetchedBalances, setHasUnfetchedBalances] = useState(true)
+  const [hasUnfetchedBalances, setHasUnfetchedBalances] = useAtom(hasUnfetchedBalancesAtom)
```

Comment:
Leave a comment explaining why an atom is necessary versus using plain state.

```suggestion
  // Use an atom to track unfetched state to avoid duplicating fetches if this component appears multiple times on the page.
  const [hasUnfetchedBalances, setHasUnfetchedBalances] = useAtom(hasUnfetchedBalancesAtom)
```